src/MongoDB.Driver.Core/Core/Compression/Zstandard/ZstandardStream.cs

```
@@ -0,0 +1,376 @@
+
+            _nativeWrapper = new ZstandardNativeWrapper(_compressionMode, EnsureCompressionLevelIsValid(compressionLevel));
+            switch (_compressionMode)
+            {
+                case CompressionMode.Compress:
+                    _streamWriteHelper = new StreamWriteHelper(
+                        compressedStream: compressedStream,
+                        compressedBuffer: _arrayPool.Rent(_nativeWrapper.RecommendedOutputSize));
+                    break;
+                case CompressionMode.Decompress:
+                    _streamReadHelper = new StreamReadHelper(
+                        compressedStream: compressedStream,
+                        compressedBuffer: _arrayPool.Rent(_nativeWrapper.RecommendedInputSize));
+                    break;
+            }
+        }
+
+        public override bool CanRead => _compressedStream.CanRead && _compressionMode == CompressionMode.Decompress;
+
+        public override bool CanWrite => _compressedStream.CanWrite && _compressionMode == CompressionMode.Compress;
+
+        public override bool CanSeek => false;
+
+        public override long Length => throw new NotSupportedException();
+
+        public override long Position
+        {
+            get => throw new NotSupportedException();
+            set => throw new NotSupportedException();
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+
+            if (!_disposed)
+            {
+                switch (_compressionMode)
+                {
+                    case CompressionMode.Compress:
+                        if (!_flushed)
+                        {
+                            try
+                            {
+                                Flush();
+                            }
+                            catch
+                            {
+                                // ignore exception
+                            }
```

There's nothing wrong with:

    try { Flush(); } catch { }

it's a well known pattern that

    try { ... } catch { }

is used to ignore exceptions. Using 8 lines instead of 1 is overkill.

---

done